salt/modules/salt_proxy.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+'''
+    Salt proxy module
+
+    .. versionadded:: 2015.8.3
+
+    Module to deploy and manage salt-proxy processes
+    on a minion.
+'''
+from __future__ import absolute_import
+
+from salt.ext.six.moves import shlex_quote as quote
+from salt.utils import fopen
+
+import os
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def _write_proxy_conf(proxyfile):
+    '''
+    write to file
+    '''
+    msg = 'Invalid value for proxy file provided!, Supplied value = {0}' \
+        .format(proxyfile)
+
+    log.trace('Salt Proxy Module: write proxy conf')
+
+    if proxyfile:
+        log.debug('Writing proxy conf file')
+        with fopen(proxyfile, 'w') as proxy_conf:
+            proxy_conf.write('master = {0}'
+                             .format(__grains__['master']))
+        msg = 'Wrote proxy file {0}'.format(proxyfile)
+        log.debug(msg)
+
+    return msg
+
+
+def _proxy_conf_file(proxyfile, test):
+    '''
+    Check if proxy conf exists and update
+    '''
+    changes_old = []
+    changes_new = []
+    success = True
+    if not os.path.exists(proxyfile):
+        try:
+            if not test:
+                changes_new.append(_write_proxy_conf(proxyfile))
+                msg = 'Salt Proxy: Wrote proxy conf {0}'.format(proxyfile)
+            else:
+                msg = 'Salt Proxy: Update required to proxy conf {0}' \
+                    .format(proxyfile)
+        except (OSError, IOError) as err:
+            success = False
+            msg = 'Salt Proxy: Error writing proxy file {0}'.format(str(err))
+            log.error(msg)
+            changes_new.append(msg)
+        changes_new.append(msg)
+        log.debug(msg)
+    else:
+        msg = 'Salt Proxy: {0} already exists, skipping'.format(proxyfile)
+        changes_old.append(msg)
+        log.debug(msg)
+    return success, changes_new, changes_old
+
+
+def _is_proxy_running(proxyname):
+    '''
+    Check if proxy for this name is running
+    '''
+    cmd = ('ps ax | grep "salt-proxy --proxyid={0}" | grep -v grep'
+           .format(proxyname))
+    cmdout = __salt__['cmd.run_all'](
+        cmd,
+        timeout=5,
+        python_shell=True)
```

Comment:
Hi @cachedout the `proxyname` variable does get quoted up in the call chain. But I am still going to change it so that we do the quote right where it is used.